src/sourmash/tax/__main__.py

```diff
@@ -0,0 +1,300 @@
+
+summarize <gather_results> [<gather_results> ... ]        - summarize taxonomic information for metagenome gather results
+classify <gather_results> [<gather_results> ... ]         - taxonomic classification of genomes from gather results
+label <gather_results> [<gather_results> ... ]            - add taxonomic information to gather results csv(s)
+
+** Use '-h' to get subcommand-specific help, e.g.
+
+sourmash taxonomy summarize -h
+'''
+
+# some utils
+def make_outfile(base, ext):
+    if base == "-":
+        return base
+    return base + ext
+
+
+##### taxonomy command line functions
+def summarize(args):
+    """
+    summarize taxonomic information for metagenome gather results
+    """
+    set_quiet(args.quiet)
+
+    # first, load taxonomic_assignments
+    tax_assign = {}
+    available_ranks = set()
+    for tax_csv in args.taxonomy_csv:
+
+        try:
+            this_tax_assign, _, avail_ranks = tax_utils.load_taxonomy_csv(tax_csv, split_identifiers=not args.keep_full_identifiers,
+                                              keep_identifier_versions = args.keep_identifier_versions,
+                                              force=args.force)
+            # maybe check for overlapping tax assignments? rn later ones will override earlier ones
+            tax_assign.update(this_tax_assign)
+            available_ranks.update(set(avail_ranks))
+
+        except ValueError as exc:
+            error(str(exc))
+
+    if not tax_assign:
+        error(f'No taxonomic assignments loaded from {args.taxonomy_csv}. Exiting.')
+        sys.exit(-1)
+
+    if args.rank and args.rank not in available_ranks:
+        error(f"No taxonomic information provided for rank {args.rank}: cannot summarize at this rank")
+        sys.exit(-1)
+
+    # next, collect and load gather results
+    gather_csvs = tax_utils.collect_gather_csvs(args.gather_csv, from_file= args.from_file)
+    gather_results, idents_missed, total_missed, _ = tax_utils.check_and_load_gather_csvs(gather_csvs, tax_assign, force=args.force,
+                                                                                       fail_on_missing_taxonomy=args.fail_on_missing_taxonomy)
+
+    if not gather_results:
+        notify(f'No gather results loaded. Exiting.')
+        sys.exit(-1)
+
+    # actually summarize at rank
+    summarized_gather = {}
+    seen_perfect = set()
+    for rank in sourmash.lca.taxlist(include_strain=False):
+        summarized_gather[rank], seen_perfect = tax_utils.summarize_gather_at(rank, tax_assign, gather_results, skip_idents=idents_missed,
+                                                                split_identifiers=not args.keep_full_identifiers,
+                                                                keep_identifier_versions = args.keep_identifier_versions,
+                                                                seen_perfect = seen_perfect)
+
+    # write summarized output csv
+    if "summary" in args.output_format:
```

Comment:
idle curiousity: what happens if there are no output formats specified? warning? error?

Reply:
the default is this `summary` (or tbd, `csv_summary`) format, so args.output_format should never be empty